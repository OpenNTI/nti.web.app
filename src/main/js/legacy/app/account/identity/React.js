const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
