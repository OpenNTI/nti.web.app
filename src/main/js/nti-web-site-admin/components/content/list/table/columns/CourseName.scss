@import '~@nti/style-common';

.content-table-container {
	.coursename-col {
		padding-left: 1rem;

		.cell {
			display: flex;

			.site-admin-course-info {
				width: 100%;

				.course-icon {
					width: 5rem;
					background-size: contain;
				}

				.info {
					max-width: 100%;

					.label {
						font-size: 10px;
					}

					.title {
						font-size: 16px;
					}

					.meta {
						display: none;
					}
				}
			}
		}
	}
}
