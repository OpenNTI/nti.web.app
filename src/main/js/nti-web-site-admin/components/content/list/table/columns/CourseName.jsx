import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';
import {CourseItem} from '@nti/web-course';

const t = scoped('nti-web-site-admin.components.content.list.table.columns.CourseName', {
	headerTitle: 'Type',
	title: 'Name'
});

export default class CourseName extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
		store: PropTypes.object.isRequired
	}

	static cssClassName = 'coursename-col';

	static Name = () => t('title')

	// static SortKey = 'Title'

	render () {
		const {item} = this.props;
		return (
			<LinkTo.Object object={item} activeClassName="active" exact context="site-admin.course-list-item">
				<div className="cell">
					<CourseItem className="course-info" catalogEntry={item.CatalogEntry} />
				</div>
			</LinkTo.Object>
		);
	}
}
