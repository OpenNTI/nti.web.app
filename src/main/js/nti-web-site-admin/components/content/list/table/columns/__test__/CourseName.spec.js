/* eslint-env jest */
import React from 'react';
import { create, act } from 'react-test-renderer';

import CourseName from '../CourseName';

describe('Site admin content table course name column test', () => {
	test('Test name', () => {
		let cmp;
		act(() => {
			cmp = create(
				<CourseName
					item={{
						CatalogEntry: {
							getStartDate: () => new Date('2/1/2017'),
							getEndDate: () => new Date('4/5/2017'),
							Title: 'Test course',
							ProviderUniqueID: 'testCourse',
							getDefaultAssetRoot() {
								return 'testRoot';
							},
						},
					}}
				/>
			);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});
});
