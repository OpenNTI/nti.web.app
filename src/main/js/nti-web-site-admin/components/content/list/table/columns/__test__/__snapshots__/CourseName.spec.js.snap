// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin content table course name column test Test name 1`] = `
<div
  className="nti-link-to-path"
  exact={true}
  to="#"
>
  <div
    className="cell"
  >
    <div
      className="site-admin-course-info course-info"
    >
      <div
        className="course-icon"
        style={
          Object {
            "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
          }
        }
      />
      <div
        className="info"
      >
        <div
          className="label"
        >
          testCourse
        </div>
        <div
          className="title"
        >
          Test course
        </div>
        <div
          className="meta"
        >
          <div
            className="start date"
          >
            <span
              className="label"
            >
              Starts:
            </span>
            <time
              className="value ntiText"
              dateTime="2017-02-01T00:00:00Z"
            >
              Feb 1, 2017
            </time>
          </div>
          <div
            className="end date"
          >
            <span
              className="label"
            >
              Ends:
            </span>
            <time
              className="value ntiText"
              dateTime="2017-04-05T00:00:00Z"
            >
              Apr 5, 2017
            </time>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
